### PR TITLE
Switch to embroider

### PR DESCRIPTION
**Pull request by mattmcmanus**
- [x] Setup embroider
- [x] Fix linking of assets in fixture data
- [x] Fix links to assets in CSS
  - WIP commit is webpack specific
  - Another option is to stick with ember-cli-postcss and try https://github.com/postcss/postcss-url
- [x] Urls for flac player
- [x] Review Ed's comments about
- [x] Fix issue with svg-jar images not working
- [x] Test production

Fixes CRD-12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,12 +18,16 @@
...
     "ember-animated": "^0.10.0",
     "ember-cli-babel": "^7.19.0",
     "ember-cli-htmlbars": "^4.2.3",
     "ember-cli-is-component": "^0.5.0",
+    "ember-cli-postcss": "^7.0.0",
     "ember-css-url": "^0.2.1",
-    "ember-element-helper": "^0.2.0"
+    "ember-element-helper": "^0.2.0",
+    "postcss-import": "^14.0.0",
+    "tailwindcss": "^2.0.2"
```
Comment (lukemelia) [inline package.json]: Did you mean to include tailwind here?

**Comment by lukemelia** [comment]
Note that this requires the following PR to embroider to be merged and released to build: https://github.com/embroider-build/embroider/pull/663

File: app/styles/app.css [hunk 2]
```diff
@@ -1,51 +1,51 @@
...
+@import '/templates/media-registry/collection.css';
+@import '/templates/media-registry/item.css';
+@import '/templates/media-registry/item-edit.css';
+@import '/templates/media-registry/agreements.css';
+@import '/templates/media-registry/breadcrumbs.css';
+@import '/templates/media-registry/embedded-collection-table.css';
+@import '/templates/media-registry/discrepancies-index.css';
+@import '/templates/media-registry/discrepancy.css';
+@import '/templates/wave-player.css';
+@import '/templates/workflow.css';
```
Comment (burieberry) [inline app/styles/app.css]: We didn't need to write any import statements with embroider in cardhost, so maybe these can be removed.

Comment (mattmcmanus) [reply app/styles/app.css]: Hey @burieberry, I'm trying to see how things may have been setup in cardhost for that to work, but I'm not finding anything. All I can find is a bunch of imports: https://github.com/cardstack/cardstack/blob/master/packages/cardhost/app/styles/app.css.

Comment (burieberry) [reply app/styles/app.css]: Oh nevermind then. Maybe I'm mixing it up with a different project.

**Comment by burieberry** [comment]
@mattmcmanus I did a general overhaul of CSS variables, so some files are conflicting here. Seems like easy fixes, but let me know if you want me to resolve these.

**Comment by mattmcmanus** [comment]
Thanks @burieberry. I'll tackle these once embroider is released.

**Comment by ef4** [comment]
@mattmcmanus here is a workaround for the test problem you asked me about:

```diff
diff --git a/tests/index.html b/tests/index.html
index 9dc212b..fb050d5 100644
--- a/tests/index.html
+++ b/tests/index.html
@@ -21,7 +21,7 @@
     {{content-for "body"}}
     {{content-for "test-body"}}

-    <script src="/testem.js" integrity=""></script>
+    <script src="{{rootURL}}testem.js" integrity=""></script>
     <script src="{{rootURL}}assets/vendor.js"></script>
     <script src="{{rootURL}}assets/test-support.js"></script>
     <script src="{{rootURL}}assets/@cardstack/boxel.js"></script>
```

I think running the test suite under a customized `rootURL` may not be a well-tested scenario in embroider.

File: package.json [hunk 3]
```diff
@@ -18,17 +18,25 @@
...
     "ember-cli-htmlbars": "^4.2.3",
     "ember-cli-is-component": "^0.5.0",
+    "ember-cli-postcss": "^7.0.0",
     "ember-css-url": "^0.2.1",
-    "ember-element-helper": "^0.2.0"
+    "ember-element-helper": "^0.2.0",
+    "postcss-import": "^14.0.0"
   },
   "devDependencies": {
+    "@babel/core": "^7.12.10",
```
Comment (ef4) [inline package.json]: The yarn warnings about this are annoying, but I worry that adding this extra unused copy just creates more opportunities for the many yarn/npm peerDep bugs to bite us. 

There are multiple layers of brokenness causing these warnings, none of which we can really fix by ourselves in this app.

I'm OK with leaving this in, but we should remember to try taking it out if we hit babel version skew problems.

**Comment by lukemelia** [comment]
@ef4 I've been working with @mattmcmanus to apply the image refactor you demo'd and complete the move to embroider. This branch now has all images referenced from JS files using `import` and that part is working nicely.

We have so far failed to achieve the same thing with images that are referenced from CSS files. After a bunch of troubleshooting and debugging, I am unclear about how CSS fits into embroider.

If I inspect the temporary directory that webpack is run against (after what I understand to be the first two stages of embroider have already run), I see our css at `TMPDIR/assets/@cardstack/boxel.css` and others at `TMPDIR/assets/components/*.css`. Any files originating `app/images` that are only referenced by CSS (such as `app/images/media-registry/placeholders/blank-card.png`) are not present in `TMPDIR/assets/images`.

I've tried configuring webpack's `url-loader` but the loader is never invoked. From my debugging, it may be this is because boxel.css is not a webpack entrypoint?

Can you provide some guidance to get us unstuck? You can pair with Matt or me if that is easiest.

**Comment by ef4** [comment]
The classic CSS pipeline in ember-cli is sufficiently convoluted that Embroider just leaves it alone. Attempts to rationalize it without breaking compatibility did not succeed. But you can stop using that legacy pipeline and directly let embroider handle the CSS.

Embroider supports declaring a dependency between a JS file and a CSS file by importing the css file:

```js
import './my-styles.css';
```

`@embroider/webpack` comes with css-loader and style-loader preconfigured to make this work. We could add postcss-loader too directly in the app if we want, although it's not clear if that's really important right now (we wouldn't need it to get the files bundled together, that is handled already by `@embroider/webpack`, so it's really only for supporting older browsers, which is a thing we could prioritize later).

File: package.json (hunk 3 above)
Comment (mattmcmanus) [reply package.json]: That's super helpful. My understanding of this level of the stack is only surface level. I'll just remove the import for now if it's not necessary.

**Comment by mattmcmanus** [comment]
@lukemelia @ef4 Hello gents, I've gone ahead done all of the imports of styles through webpack. You can see it in d4d6a96.

Things that are still not super clear to me: 

* `url()`s in stylesheets are still untouched
* Importing css through JS is a big mental model shift. Are the advantages exclusively around codesplitting?
* The css imports needed to have `/assets/` instead of styles. I'm assuming that this is because styles are copied over in some part of the build process. It was non-obvious to try and figure out why it was `assets` and not `styles` 
* I'm still struggling with the mental model of where the boundaries between ember-cli/embroider/webpack are. Since there is already some style processing happen, it's not clear to me where to start setting up post-css to use plugins like https://github.com/postcss/postcss-nested

Hopefully this makes sense. I'm happy to pair and talk through some things or be pointed in the direction of a talk or something that may help.

**Comment by lukemelia** [comment]
I pushed 6b3dfb26b7a8093b471ebc95776fceb111cf499f, which fixes the issue with `url(...)`s in stylesheets not being processed. In fact they were being considered, but only relative URLs are processed and almost all started with a "/".

The confusion around when to use what paths to reference an image is not quite as bad now. One remaining point of vonfusion is that an image on disk at `app/images/foo.jpg` would be imported in javascript as `@cardstack/boxel/images/foo.jpg` but in css only a relative path is acceptable.

Regarding adding a postcss loader, a few thoughts:

1) We can now get colocated component css just by moving the file to the component dir and importing it from the component's index.js. So we don't need postcss-import.
2) postcss-nested is not desirable IMO because css in cards needs to be pasted into a browser editor and is not processed, so it is advantageous for Boxel source to authored in a similar way.

So I don't think we need to deal with this for now. I am still curious about the relationship between loaders configured in ember-cli-build's packagerOptions and the ones configured within embroider/webpack. Perhaps @ef4 can explain.

**Comment by ef4** [comment]
> One remaining point of confusion is that an image on disk at app/images/foo.jpg would be imported in javascript as @cardstack/boxel/images/foo.jpg but in css only a relative path is acceptable.

This is because in the Javascript, Embroider is rewriting package self-imports into relative imports before handing off to webpack. That's a compatibility feature since traditional Ember apps do that a lot. Webpack itself is following node_modules resolution rules, which don't normally support packages resolving themselves.

**Comment by mattmcmanus** [comment]
I'm running into issues trying to get this to complete a `-prod` build. The error is: 

```
 - broccoliBuilderErrorStack: Error: needed by index.html: ENOENT: no such file or directory, open '/var/folders/f8/xfnrq0_13ps4cr_dmy07306m0000gn/T/broccoli-68654gA7b0qUwSG2R/out-467-packager_runner_embroider_webpack/assets/@cardstack/boxel.d41d8cd98f00b204e9800998ecf8427e.css'
    at Object.openSync (fs.js:462:3)
    at Object.writeFileSync (fs.js:1384:35)
    at Webpack.writeStyle (/Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:229:24)
```
<details>
  <summary>Full crash dump</summary>

```
=================================================================================

ENV Summary:

  TIME: Thu Jan 14 2021 14:59:17 GMT-0500 (Eastern Standard Time)
  TITLE: ember
  ARGV:
  - /Users/matt/.volta/tools/image/node/12.20.1/bin/node
  - /Users/matt/p/boxel/node_modules/.bin/ember
  - s
  - -prod
  EXEC_PATH: /Users/matt/.volta/tools/image/node/12.20.1/bin/node
  TMPDIR: /var/folders/f8/xfnrq0_13ps4cr_dmy07306m0000gn/T
  SHELL: /bin/bash
  PATH:
  - /var/folders/f8/xfnrq0_13ps4cr_dmy07306m0000gn/T/yarn--1610654039243-0.6001531820917967
  - /Users/matt/p/boxel/node_modules/.bin
  - /Users/matt/.config/yarn/link/node_modules/.bin
  - /Users/matt/.volta/tools/image/node/12.20.1/libexec/lib/node_modules/npm/bin/node-gyp-bin
  - /Users/matt/.volta/tools/image/node/12.20.1/lib/node_modules/npm/bin/node-gyp-bin
  - /Users/matt/.volta/tools/image/node/12.20.1/bin/node_modules/npm/bin/node-gyp-bin
  - /Users/matt/.volta/tools/image/yarn/1.22.10/bin
  - /Users/matt/.volta/tools/image/node/12.20.1/bin
  - /Users/matt/.volta/bin
  - /Users/matt/.pyenv/shims
  - /Users/matt/.nodenv/shims
  - /Users/matt/.rbenv/shims
  - /usr/local/opt/postgresql@9.5/bin
  - /Users/matt/.yarn/bin
  - /usr/local/sbin
  - /usr/local/bin
  - /Users/matt/local/bin
  - ~/Library/Android/sdk/tools/
  - /usr/local/heroku/bin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/sbin
  - /sbin
  - /usr/local/MacGPG2/bin
  - /Library/Apple/usr/bin
  - /Applications/Wireshark.app/Contents/MacOS
  - /Users/matt/.volta/bin
  - /Users/matt/.pyenv/shims
  - /Users/matt/.nodenv/shims
  - /Users/matt/.rbenv/shims
  - /usr/local/opt/postgresql@9.5/bin
  - /Users/matt/.yarn/bin
  - /usr/local/sbin
  - /Users/matt/local/bin
  - ~/Library/Android/sdk/tools/
  - /usr/local/heroku/bin
  - /Users/matt/.dotfiles/bin
  - /Users/matt/.dotfiles/bin
  PLATFORM: darwin x64
  FREEMEM: 354041856
  TOTALMEM: 34359738368
  UPTIME: 65025
  LOADAVG: 3.72802734375,3.86669921875,3.61865234375
  CPUS:
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  - Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz - 2900
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.16.1
  - brotli: 1.0.9
  - cldr: 37.0
  - http_parser: 2.9.4
  - icu: 67.1
  - llhttp: 2.1.3
  - modules: 72
  - napi: 7
  - nghttp2: 1.41.0
  - node: 12.20.1
  - openssl: 1.1.1i
  - tz: 2019c
  - unicode: 13.0
  - uv: 1.40.0
  - v8: 7.8.279.23-node.45
  - zlib: 1.2.11

ERROR Summary:

  - broccoliBuilderErrorStack: Error: needed by index.html: ENOENT: no such file or directory, open '/var/folders/f8/xfnrq0_13ps4cr_dmy07306m0000gn/T/broccoli-68654gA7b0qUwSG2R/out-467-packager_runner_embroider_webpack/assets/@cardstack/boxel.d41d8cd98f00b204e9800998ecf8427e.css'
    at Object.openSync (fs.js:462:3)
    at Object.writeFileSync (fs.js:1384:35)
    at Webpack.writeStyle (/Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:229:24)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:276:76
    at async Webpack.provideErrorContext (/Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:235:24)
    at async Webpack.writeFiles (/Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:251:17)
    at async Webpack.build (/Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:77:13)
    at async TransformNodeWrapper.build (/Users/matt/p/boxel/node_modules/broccoli/dist/wrappers/transform-node.js:70:13)
    at async /Users/matt/p/boxel/node_modules/broccoli/dist/builder.js:118:21
    at async Builder.build (/Users/matt/p/boxel/node_modules/broccoli/dist/builder.js:137:13)
  - code: [undefined]
  - codeFrame: needed by index.html: ENOENT: no such file or directory, open '/var/folders/f8/xfnrq0_13ps4cr_dmy07306m0000gn/T/broccoli-68654gA7b0qUwSG2R/out-467-packager_runner_embroider_webpack/assets/@cardstack/boxel.d41d8cd98f00b204e9800998ecf8427e.css'
  - errorMessage: needed by index.html: ENOENT: no such file or directory, open '/var/folders/f8/xfnrq0_13ps4cr_dmy07306m0000gn/T/broccoli-68654gA7b0qUwSG2R/out-467-packager_runner_embroider_webpack/assets/@cardstack/boxel.d41d8cd98f00b204e9800998ecf8427e.css'
        at PackagerRunner (@embroider/webpack)
-~- created here: -~-
    at new Plugin (/Users/matt/p/boxel/node_modules/@embroider/core/node_modules/broccoli-plugin/dist/index.js:45:33)
    at new PackagerRunner (/Users/matt/p/boxel/node_modules/@embroider/core/src/to-broccoli-plugin.js:10:13)
    at Object.defaultPipeline (/Users/matt/p/boxel/node_modules/@embroider/compat/src/default-pipeline.js:43:12)
    at module.exports (/Users/matt/p/boxel/ember-cli-build.js:29:39)
    at Builder.readBuildFile (/Users/matt/p/boxel/node_modules/ember-cli/lib/models/builder.js:49:14)
    at Builder.setupBroccoliBuilder (/Users/matt/p/boxel/node_modules/ember-cli/lib/models/builder.js:63:22)
    at new Builder (/Users/matt/p/boxel/node_modules/ember-cli/lib/models/builder.js:29:10)
    at ServeTask.run (/Users/matt/p/boxel/node_modules/ember-cli/lib/tasks/serve.js:49:7)
    at /Users/matt/p/boxel/node_modules/ember-cli/lib/models/command.js:238:24
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
-~- (end) -~-
  - errorType: Build Error
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
    - treeDir: [undefined]
  - message: needed by index.html: ENOENT: no such file or directory, open '/var/folders/f8/xfnrq0_13ps4cr_dmy07306m0000gn/T/broccoli-68654gA7b0qUwSG2R/out-467-packager_runner_embroider_webpack/assets/@cardstack/boxel.d41d8cd98f00b204e9800998ecf8427e.css'
        at PackagerRunner (@embroider/webpack)
-~- created here: -~-
    at new Plugin (/Users/matt/p/boxel/node_modules/@embroider/core/node_modules/broccoli-plugin/dist/index.js:45:33)
    at new PackagerRunner (/Users/matt/p/boxel/node_modules/@embroider/core/src/to-broccoli-plugin.js:10:13)
    at Object.defaultPipeline (/Users/matt/p/boxel/node_modules/@embroider/compat/src/default-pipeline.js:43:12)
    at module.exports (/Users/matt/p/boxel/ember-cli-build.js:29:39)
    at Builder.readBuildFile (/Users/matt/p/boxel/node_modules/ember-cli/lib/models/builder.js:49:14)
    at Builder.setupBroccoliBuilder (/Users/matt/p/boxel/node_modules/ember-cli/lib/models/builder.js:63:22)
    at new Builder (/Users/matt/p/boxel/node_modules/ember-cli/lib/models/builder.js:29:10)
    at ServeTask.run (/Users/matt/p/boxel/node_modules/ember-cli/lib/tasks/serve.js:49:7)
    at /Users/matt/p/boxel/node_modules/ember-cli/lib/models/command.js:238:24
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
-~- (end) -~-
  - name: Error
  - nodeAnnotation: @embroider/webpack
  - nodeName: PackagerRunner
  - originalErrorMessage: needed by index.html: ENOENT: no such file or directory, open '/var/folders/f8/xfnrq0_13ps4cr_dmy07306m0000gn/T/broccoli-68654gA7b0qUwSG2R/out-467-packager_runner_embroider_webpack/assets/@cardstack/boxel.d41d8cd98f00b204e9800998ecf8427e.css'
  - stack: Error: needed by index.html: ENOENT: no such file or directory, open '/var/folders/f8/xfnrq0_13ps4cr_dmy07306m0000gn/T/broccoli-68654gA7b0qUwSG2R/out-467-packager_runner_embroider_webpack/assets/@cardstack/boxel.d41d8cd98f00b204e9800998ecf8427e.css'
    at Object.openSync (fs.js:462:3)
    at Object.writeFileSync (fs.js:1384:35)
    at Webpack.writeStyle (/Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:229:24)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:276:76
    at async Webpack.provideErrorContext (/Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:235:24)
    at async Webpack.writeFiles (/Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:251:17)
    at async Webpack.build (/Users/matt/p/boxel/node_modules/@embroider/webpack/src/ember-webpack.js:77:13)
    at async TransformNodeWrapper.build (/Users/matt/p/boxel/node_modules/broccoli/dist/wrappers/transform-node.js:70:13)
    at async /Users/matt/p/boxel/node_modules/broccoli/dist/builder.js:118:21
    at async Builder.build (/Users/matt/p/boxel/node_modules/broccoli/dist/builder.js:137:13)

=================================================================================
```
</details>

The tmp directory doesn't have any app css in it:
Screen Shot 2021-01-14 at 3 20 25 PM-399x309

I'm going to keep poking around, but any insight you may have would be helpful.

**Comment by ef4** [comment]
Make sure any CSS that you are trying to import (the "new way") is not under app/styles (because that all gets handled the "old way".

**Comment by mattmcmanus** [comment]
@ef4 Ok. So I've moved everything out of `app/styles` to `app/css`. Things still seem to work fine under dev mode.

<details>
  <summary>Dev mode details and layout</summary>

## Dist layout
Screen Shot 2021-01-14 at 4 50 12 PM-559x687


## Generated index.html
```
<!DOCTYPE html><html lang="en"><head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Boxel</title>
    <meta name="description" content="">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    
<meta name="@cardstack/boxel/config/environment" content="%7B%22modulePrefix%22%3A%22%40cardstack%2Fboxel%22%2C%22environment%22%3A%22development%22%2C%22rootURL%22%3A%22%2F%22%2C%22locationType%22%3A%22auto%22%2C%22EmberENV%22%3A%7B%22FEATURES%22%3A%7B%22EMBER_NATIVE_DECORATOR_SUPPORT%22%3Atrue%2C%22EMBER_METAL_TRACKED_PROPERTIES%22%3Atrue%2C%22EMBER_GLIMMER_ANGLE_BRACKET_NESTED_LOOKUP%22%3Atrue%2C%22EMBER_GLIMMER_ANGLE_BRACKET_BUILT_INS%22%3Atrue%2C%22EMBER_GLIMMER_FN_HELPER%22%3Atrue%2C%22EMBER_GLIMMER_ON_MODIFIER%22%3Atrue%7D%2C%22EXTEND_PROTOTYPES%22%3A%7B%22Date%22%3Afalse%7D%2C%22_APPLICATION_TEMPLATE_WRAPPER%22%3Afalse%2C%22_JQUERY_INTEGRATION%22%3Atrue%2C%22_TEMPLATE_ONLY_GLIMMER_COMPONENTS%22%3Atrue%7D%2C%22APP%22%3A%7B%7D%2C%22exportApplicationGlobal%22%3Atrue%2C%22ember-faker%22%3A%7B%7D%2C%22ember-a11y-testing%22%3A%7B%22componentOptions%22%3A%7B%22turnAuditOff%22%3Atrue%2C%22visualNoiseLevel%22%3A2%2C%22axeViolationClassNames%22%3A%5B%22alert-box%22%2C%22alert-box--a11y%22%5D%7D%7D%7D">
<script src="/ember-cli-live-reload.js" type="text/javascript"></script>

    
<link href="/assets/vendor.css" rel="stylesheet">

    
<link href="/assets/@cardstack/boxel.css" rel="stylesheet">


    
  </head>
  <body>
    

    
<script src="/assets/vendor.js"></script>

    
<script src="/assets/chunk.a75198b77c6470ce3833.js"></script>
<script src="/assets/chunk.bdf950899b95a4ee6607.js"></script>
<script src="/assets/chunk.564eefe3ff9d2724fe09.js"></script>


    <div id="ember-basic-dropdown-wormhole"></div>
  

</body></html>
```
</details>

But in production, it now successfully complete the build, but does some strange things. In particular, it seems to be writing what should be the application js as the css styles. 

Here is the generated index.html with some comments

```html
<!DOCTYPE html><html lang="en"><head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Boxel</title>
    <meta name="description" content="">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    
<meta name="@cardstack/boxel/config/environment" content="%7B%22modulePrefix%22%3A%22%40cardstack%2Fboxel%22%2C%22environment%22%3A%22production%22%2C%22rootURL%22%3A%22%2Fboxel%2F%22%2C%22locationType%22%3A%22hash%22%2C%22EmberENV%22%3A%7B%22FEATURES%22%3A%7B%22EMBER_NATIVE_DECORATOR_SUPPORT%22%3Atrue%2C%22EMBER_METAL_TRACKED_PROPERTIES%22%3Atrue%2C%22EMBER_GLIMMER_ANGLE_BRACKET_NESTED_LOOKUP%22%3Atrue%2C%22EMBER_GLIMMER_ANGLE_BRACKET_BUILT_INS%22%3Atrue%2C%22EMBER_GLIMMER_FN_HELPER%22%3Atrue%2C%22EMBER_GLIMMER_ON_MODIFIER%22%3Atrue%7D%2C%22EXTEND_PROTOTYPES%22%3A%7B%22Date%22%3Afalse%7D%2C%22_APPLICATION_TEMPLATE_WRAPPER%22%3Afalse%2C%22_JQUERY_INTEGRATION%22%3Atrue%2C%22_TEMPLATE_ONLY_GLIMMER_COMPONENTS%22%3Atrue%7D%2C%22APP%22%3A%7B%7D%2C%22exportApplicationGlobal%22%3Afalse%2C%22ember-faker%22%3A%7B%7D%7D">
<script src="/boxel/ember-cli-live-reload.js" type="text/javascript"></script>

    
<link href="/boxel/assets/vendor.067892fe703489669be9ec58eb04fc23.css" rel="stylesheet">

    
<link rel="stylesheet" href="/boxel/assets/@cardstack/boxel.css"> <!-- THIS file does not exist in Dist -->

    
  </head>
  <body>
    

    
<script src="/boxel/assets/vendor.72697a1b37bba6c79b2a38d387ef8bd5.js"></script>

    
<script src="/boxel/assets/chunk.9d661ae73736a9648bf6.js"></script>
<script src="/boxel/assets/assets/@cardstack/boxel.js.css"></script> <!-- This is what should be the css, but in script tag -->
<script src="/boxel/assets/chunk.6e46f9d6b0955d78dc89.js"></script>


    <div id="ember-basic-dropdown-wormhole"></div>
  

</body></html>
```
There is also some strange things going on with the assets directory, ie: assets/assets/
Screen Shot 2021-01-14 at 4 59 39 PM-404x449

It also is strange to me that things aren't fingerprinted. 

Seems like the bug is somewhere in https://github.com/embroider-build/embroider/blob/master/packages/webpack/src/ember-webpack.ts#L264. I'll try to run through this with an inspector attached tomorrow.

**Comment by lukemelia** [comment]
Waiting for this to be merged and released: https://github.com/embroider-build/embroider/pull/670

**Comment by mattmcmanus** [comment]
@burieberry This PR is ready for merging and review. It has some big changes. I made a 4 minutes video quickly going over them. I'm happy to setup a time to talk as well. Just let me know. 

https://www.loom.com/share/55e2d5ad64dc422bb3aa3d293e5daaa0

**Comment by burieberry** [comment]
> @burieberry This PR is ready for merging and review. It has some big changes. I made a 4 minutes video quickly going over them. I'm happy to setup a time to talk as well. Just let me know.
> 
> https://www.loom.com/share/55e2d5ad64dc422bb3aa3d293e5daaa0

This is very helpful and clear, thank you. The directory structure looks good to me, and did a quick click-through on the app and everything seems good.

**Review by burieberry — APPROVED** [review]

**Pull request merged**